### PR TITLE
[CN-733] Run the terraform-integration-tests by PR label event

### PR DESCRIPTION
**Pull request by semihbkgr**
Triggering the discovery plugin tests workflows in a PR which is labeled with a specific label.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by mtyazici — COMMENTED** [review]
You should update the checkout actions. Now they will checkout to the master branch. However, we should checkout to the PR merge branch.

File: .github/workflows/aws-terraform-integration-tests.yml [hunk 1]
```diff
@@ -32,12 +48,16 @@ jobs:
...
+      - name: JDK info
+        run: java -version
 
       # BUILD HAZELCAST AWS SNAPSHOT
-      - uses: actions/checkout@v3.3.0
+      - name: Checkout
+        uses: actions/checkout@v3.3.0
         with:
           path: hazelcast
+          ref: ${{steps.decide-ref.outputs.ref}}
```
Comment (mtyazici) [inline .github/workflows/aws-terraform-integration-tests.yml:59]: We do not have a decide-ref step here if I am not wrong, could you provide it?

Comment (semihbkgr) [reply .github/workflows/aws-terraform-integration-tests.yml:59]: You mean use only `ref: ${{ github.ref }}`

Comment (mtyazici) [reply .github/workflows/aws-terraform-integration-tests.yml:59]: No, like this https://github.com/hazelcast/hazelcast-platform-operator/blob/main/.github/workflows/pull-request.yml#L220-L224

**Review by mtyazici — APPROVED** [review]
LGTM 👍🏼, maybe we could find a more descriptive label name.

I am also not sure if we still need the push trigger, should we keep it?

**Comment by semihbkgr** [comment]
I think so, I will update to `run-terraform-tests` label name what we decided.
I agree that we don't need to run tests on push trigger anymore.

**Review by hasancelik — APPROVED** [review]

**Review by mtyazici — APPROVED** [review]
LGTM

**Pull request merged**